src/theories/Category.jl

```diff
@@ -197,3 +198,22 @@ abstract type TightExpr{T} <: GATExpr{T} end
   compose(f::Hom, g::Hom) = associate_unit(new(f,g; strict=true), id)
   transitive(t::Tight, u::Tight) = associate_unit(new(t,u; strict=true), reflexive)
 end
+
+"""
+Theory of a pointed category.
+We axiomatize a category equipped with zero morphisms.
```

Comment:
Since this is going into `Catlab.Theories` I am going to bikeshed the terminology. The nLab suggests that a [pointed category](https://ncatlab.org/nlab/show/pointed+category) is a category with a zero object, whereas this is a category with [zero morphisms](https://ncatlab.org/nlab/show/zero+morphism), which is a category enriched in pointed sets.

We need a clear and not-to-long name for a "category enriched in pointed sets." `PointedSetCategory`?

Reply:
Oh, yeah, I was initially providing the zero object as well as the zero morphisms but then didn't need them. I would prefer `Set_*-Category` in principle, but `PointedSet-Category` is probably clearer to more people. On the other hand perhaps I should just put the zero objects back in so that we can have object-partial functors too for later?